packages/components/src/components/tag/tag-styles.ts

```diff
@@ -116,14 +113,11 @@ export const getComponentCss = (
         },
       }),
     },
-    icon: {
-      marginInlineStart: '-2px', // compensate white space of svg icon and optimize visual alignment
-      alignSelf: 'flex-start',
-      ...(['neutral-contrast-high', 'primary'].includes(tagColor) && {
-        ...prefersColorSchemeDarkMediaQuery(theme, {
-          filter: 'invert(1)',
-        }),
-      }),
-    },
+    ...(hasIcon && {
+      icon: {
+        marginInlineStart: '-2px', // compensate white space of svg icon and optimize visual alignment
+        filter: !isHighContrastMode && tagColor === 'primary' ? 'invert(1)' : null,
```

Comment:
```suggestion
        ...(!isHighContrastMode && tagColor === 'primary' && { filter: 'invert(1)' }),
```
Maybe better than applying null to filter